File: src/leetcode/problem/p0/P724.java
package leetcode.problem.p0;

import java.util.Arrays;

public class P724 {

    public static void main(String[] args) {
        System.out.println(new P724().pivotIndex(new int[] { 0, 0 }));
        System.out.println(new P724().pivotIndex(new int[] { 1, 7, 1 }));
        System.out.println(new P724().pivotIndex(new int[] { 1, 7, 3, 6, 5, 6 }));
        System.out.println(new P724().pivotIndex(new int[] { 1, 7, 4 }));
    }

    public int pivotIndex(int[] nums) {
        final int length = nums.length;
        final int sum = Arrays.stream(nums).sum();
        int leftSum = 0;
        int rightSum = 0;
        for (int i = 0; i < length; i++) {
            rightSum = sum - leftSum - nums[i];
            if (leftSum == rightSum) {
                return i;
            }
            leftSum += nums[i];
        }
        return -1;
    }
}

File: src/leetcode/problem/p0/P724.kt
package leetcode.problem.p0

fun main() {
    println(pivotIndex(intArrayOf(0, 0)))
    println(pivotIndex(intArrayOf(1, 7, 1)))
    println(pivotIndex(intArrayOf(1, 7, 3, 6, 5, 6)))
    println(pivotIndex(intArrayOf(1, 7, 4)))
}

fun pivotIndex(nums: IntArray): Int {
    val sum = nums.sum()
    var leftSum = 0
    var rightSum = 0

    nums.forEachIndexed{i, element ->
        rightSum = sum - leftSum - element
        if (leftSum == rightSum) {
            return i;
        }
        leftSum += element
    }

//    for ((i) in nums.withIndex()) {
//        rightSum = sum - leftSum - nums[i]
//        if (leftSum == rightSum) {
//            return i;
//        }
//        leftSum += nums[i]
//    }
    return -1
}
